DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UISwitchRecorder.swift

```diff
@@ -48,11 +49,41 @@ internal struct UISwitchWireframesBuilder: NodeWireframesBuilder {
     let isEnabled: Bool
     let isDarkMode: Bool
     let isOn: Bool
+    let isMasked: Bool
     let thumbTintColor: CGColor?
     let onTintColor: CGColor?
     let offTintColor: CGColor?
 
     func buildWireframes(with builder: WireframesBuilder) -> [SRWireframe] {
+        if isMasked {
+            return createMasked(with: builder)
+        } else {
+            return createNotMasked(with: builder)
+        }
+    }
+
+    private func createMasked(with builder: WireframesBuilder) -> [SRWireframe] {
```

Comment:
Instead of adding new method, we could patch the previous one with `if isMasked` in several places. I tried this, but after all it was less readable and coherent than here.